sparse/client.go

```diff
@@ -74,65 +88,74 @@ func SyncFile(localPath string, remote string, timeout int, directIO bool) error
 	}
 	defer fileIo.Close()
 
-	client := newSyncClient(remote, timeout, localPath, fileSize, fileIo)
+	return SyncContent(fileIo.Name(), SourceTypeFile, fileIo, fileSize, remote, timeout, directIO)
+}
+
+func SyncContent(sourceName string, sourceType SourceType, rw ReaderWriterAt, size int64, remote string, timeout int, directIO bool) (err error) {
+	defer func() {
+		err = errorWrapf(err, "failed to SyncContent for source: %v, type: %v", sourceName, sourceType)
```

Comment:
You can use `errors.Wrapf()` instead

Reply:
I was hesitated to use `errors.Wrapf()` since we would have to add go module `github.com/pkg/errors` just for the function. But using standard function may be good in the long run so I changed it as your suggestion. Thanks